internal/kafka/command_topic_produce.go

```diff
@@ -89,6 +89,15 @@ func (c *command) produce(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	parseKey, err := cmd.Flags().GetBool("parse-key")
+	if err != nil {
+		return err
+	}
+
+	if cmd.Flags().Changed("key-format") && !parseKey {
+		return errors.New("`--parse-key` must be set when `key-format` is set")
```

Comment:
how about the other way, if parse-key is set?

Reply:
That actually works, it defaults to assuming string keys. That's why I went with this instead of just marking them as required together.